apiserver/facades/agent/credentialvalidator/credentialvalidator.go

```diff
@@ -172,6 +208,16 @@ func (api *CredentialValidatorAPI) modelCredential(ctx context.Context) (*ModelC
 		return result, nil
 	}
 
+	modelCredentialKey := credential.Key{
+		Cloud: modelInfo.Cloud,
+		Owner: modelInfo.CredentialOwner,
+		Name:  modelInfo.CredentialName,
+	}
+	modelCredentialTag, err := modelCredentialKey.Tag()
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
```

Comment:
No need for this. We have the Key above already. This is making a key, making a tag from the key, and then calling a helper to get the same key back again here:
`credential, err := api.credentialService.CloudCredential(ctx, credential.KeyFromTag(modelCredentialTag))`

Reply:
Hi Ian, yup I have removed credential.KeyFromTag thank you! Please CMIIW but I think we would still need the credentialTag to be passed into params.ModelCredential in the end isn't it? So we would still need the block here.

Reply:
Yup I have removed credential.KeyFromTag thanks Ian!